rdr_service/offline/update_ehr_status.py

```diff
@@ -59,11 +61,25 @@ def update_participant_summaries_from_job(job):
         total_rows = query_result.rowcount
         LOG.info("Affected {} rows.".format(total_rows))
         if total_rows > 0:
+
             count = int(math.ceil(float(total_rows) / float(batch_size)))
             LOG.info('UpdateEhrStatus: calculated {0} participant rebuild tasks from {1} records and batch size of {2}'.
-                         format(count, total_rows, batch_size))
+                     format(count, total_rows, batch_size))
             pids = [param['pid'] for param in parameter_sets]
-            dispatch_participant_rebuild_tasks(pids, batch_size=batch_size)
+
+            with summary_dao.session() as session:
+                cursor = session.query(ParticipantSummary.participantId, ParticipantSummary.ehrReceiptTime).all()
+                records = [r for r in cursor if r.participantId in pids]
+
+            patch_data = [{
+                'pid': rec.participantId,
+                'patch': {
+                    'ehr_status': str(EhrStatus.PRESENT),
+                    'ehr_status_id': int(EhrStatus.PRESENT),
+                    'ehr_receipt': rec.ehrReceiptTime if rec.ehrReceiptTime else now,
```

Comment:
Is always using rec.ehrReceiptTime appropriate?  Theoretically, if we're here, unless the bulk update failed this field in participant summary should always be populated for all the pids this cron job processed.

If there's an RDR bug/glitch where it's null for some reason, is it preferable to populate the value for PDR anyway?  (RDR/PDR would be out of sync but maybe that's intentional).

Reply:
I think it would get corrected later on if there was an issue.